src/NUnitFramework/tests/Constraints/DictionaryContainsKeyConstraintTests.cs

```diff
@@ -84,35 +85,521 @@ public void WorksWithNonGenericDictionary()
         }
 #endif
 
-        [Test]
+        [Test, Obsolete]
         public void IgnoreCaseIsHonored()
         {
             var dictionary = new Dictionary<string, string> { { "Hello", "World" }, { "Hola", "Mundo" } };
 
             Assert.That(dictionary, new DictionaryContainsKeyConstraint("HELLO").IgnoreCase);
         }
 
-        [Test]
+        [Test, Obsolete]
         public void UsingIsHonored()
         {
             var dictionary = new Dictionary<string, string> { { "Hello", "World" }, { "Hola", "Mundo" } };
 
-            Assert.That(dictionary,
-                new DictionaryContainsKeyConstraint("HELLO").Using<string>((x, y) => StringUtil.Compare(x, y, true)));
+            Assert.That(dictionary, new DictionaryContainsKeyConstraint("HELLO").Using<string>((x, y) => StringUtil.Compare(x, y, true)));
         }
 
-        [Test, Explicit("Demonstrates issue #2837")]
+        [Test]
         public void SucceedsWhenKeyIsPresentWhenDictionaryUsingCustomComparer()
         {
             var dictionary = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase) { { "Hello", "World" }, { "Hola", "Mundo" } };
 
             Assert.That(dictionary, new DictionaryContainsKeyConstraint("hello"));
         }
-        [Test, Explicit("Demonstrates issue #2837")]
+
+        [Test]
         public void SucceedsWhenKeyIsPresentUsingContainKeyWhenDictionaryUsingCustomComparer()
         {
             var dictionary = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase) { { "Hello", "World" }, { "Hola", "Mundo" } };
             Assert.That(dictionary, Does.ContainKey("hola"));
         }
+
+        [Test]
+        public void SucceedsWhenKeyIsPresentUsingContainKeyWhenUsingLookupCustomComparer()
+        {
+            var list = new List<string> { "ALICE", "BOB", "CATHERINE" };
+            ILookup<string, string> lookup = list.ToLookup(x => x, StringComparer.OrdinalIgnoreCase);
+
+            Assert.That(lookup, Does.ContainKey("catherine"));
+        }
+
+        [Test]
+        public void SucceedsWhenKeyIsNotPresentUsingContainKeyUsingLookupDefaultComparer()
+        {
+            var list = new List<string> { "ALICE", "BOB", "CATHERINE" };
+            ILookup<string, string> lookup = list.ToLookup(x => x);
+
+            Assert.That(lookup, !Does.ContainKey("alice"));
```

Comment:
I did not know we declared a custom `!` operator! 😆

Reply:
Wow, me neither! 😄